src/main/java/org/jvnet/hudson/plugins/m2release/M2ReleaseAction.java

```diff
@@ -307,15 +306,87 @@ public ParameterDefinition getParameterDefinition(String name) {
 	}
 
 	/**
-	 * returns the value of the key as a String. if multiple values have been
-	 * submitted, the first one will be returned.
-	 * 
-	 * @param key
-	 * @param httpParams
-	 * @return
+	 * Wrapper to access request data with a special treatment if POST is multipart encoded
 	 */
-	private String getString(String key, Map<?, ?> httpParams) {
-		return (String) (((Object[]) httpParams.get(key))[0]);
+	static class StaplerRequestWrapper {
+		private final StaplerRequest request;
+		private Map<String, FileItem> parsedFormData;
+		private boolean isMultipartEncoded;
+
+		public StaplerRequestWrapper(StaplerRequest request) throws ServletException {
+			this.request = request;
+
+			// JENKINS-16043, POST can be multipart encoded if there's a file parameter in the job
+			String ct = request.getContentType();
+			if (ct != null && ct.startsWith("multipart/")) {
+				// as multipart content can only be read once, we can't read it here, otherwise it would
+				// break request.getSubmittedForm(). So, we get it using reflection by reading private
+				// field parsedFormData
+
+				// ensure parsedFormData field is filled
+				request.getSubmittedForm();
+
+				try {
+					java.lang.reflect.Field privateField = org.kohsuke.stapler.RequestImpl.class.getDeclaredField("parsedFormData");
+					privateField.setAccessible(true);
+					parsedFormData = (Map<String, FileItem>) privateField.get(request);
+				} catch (NoSuchFieldException e) {
+					throw new IllegalArgumentException(e);
+				} catch (IllegalAccessException e) {
+					throw new IllegalArgumentException(e);
+				}
+
+				isMultipartEncoded = true;
+			} else {
+				isMultipartEncoded = false;
+			}
+		}
+
+		/**
+		 * returns the value of the key as a String. if multiple values have been
+		 * submitted, the first one will be returned.
+		 *
+		 * @param key
+		 * @return
+		 */
+		private String getString(String key) throws javax.servlet.ServletException, java.io.IOException {
+			if (isMultipartEncoded) {
+				// borrowed from org.kohsuke.staple.RequestImpl
+				FileItem item = parsedFormData.get(key);
+				if (item!=null && item.isFormField()) {
+					if (item.getContentType() == null && request.getCharacterEncoding() != null) {
+						// JENKINS-11543: If client doesn't set charset per part, use request encoding
+						try {
+							return item.getString(request.getCharacterEncoding());
+						} catch (java.io.UnsupportedEncodingException uee) {
+							LOGGER.log(Level.WARNING, "Request has unsupported charset, using default for '"+key+"' parameter", uee);
+							return item.getString();
+						}
+					} else {
+						return item.getString();
+					}
+				} else {
+					throw new IllegalArgumentException("Parameter not found: " + key);
```

Comment:
previously a get on a non existent key returned `null` - the code used to make use of that probably doesn't at the moment but would be better to return `null` just in case.